packages/expo-application/src/Application.ts

```diff
@@ -3,6 +3,14 @@ import { Platform, UnavailabilityError } from 'expo-modules-core';
 import { ApplicationReleaseType, PushNotificationServiceEnvironment } from './Application.types';
 import ExpoApplication from './ExpoApplication';
 
+type JSONValue = boolean | number | string | null | JSONArray | JSONObject;
+
+interface JSONArray extends Array<JSONValue> {}
+
+interface JSONObject {
+  [key: string]: JSONValue | undefined;
+}
```

Comment:
```suggestion
type JSONArray = JSONValue[];

type JSONObject = Record<JSONValue | undefined>
```

In general, in SDK package source we prefer `types` over `interfaces` to enable `type` imports and better docs readability.

Reply:
Using `interface` for this used to be necessary. TypeScript might have changed this but previously `interface JSONArray extends Array<JSONValue> {}` was key.